ingester/src/test_util.rs

```diff
@@ -590,27 +586,38 @@ pub const TEST_PARTITION_1: &str = "test+partition_1";
 ///
 pub const TEST_PARTITION_2: &str = "test+partition_2";
 
-/// Make the same in-memory data but data are split between:
-///    . one or two partition
-///    . The first partition will have a choice to have data in either
-///       . buffer only
-///       . snapshot only
-///       . persisting only
-///       . buffer + snapshot
-///       . buffer + persisting
-///       . snapshot + persisting
-///       . buffer + snapshot + persisting
-///    . If the second partittion exists, it only has data in its buffer
-#[allow(missing_docs)]
-#[derive(Debug, PartialEq, Copy, Clone)]
-pub enum DataLocation {
-    Buffer,
-    Snapshot,
-    Persisting,
-    BufferSnapshot,
-    BufferPersisting,
-    SnapshotPersisting,
-    BufferSnapshotPersisting,
+#[cfg(test)]
+use bitflags::bitflags;
+
+#[cfg(test)]
+bitflags! {
+    /// Make the same in-memory data but data are split between:
+    ///    . one or two partition
+    ///    . The first partition will have a choice to have data in either
+    ///       . buffer only
+    ///       . snapshot only
+    ///       . persisting only
+    ///       . buffer + snapshot
+    ///       . buffer + persisting
+    ///       . snapshot + persisting
+    ///       . buffer + snapshot + persisting
+    ///    . If the second partittion exists, it only has data in its buffer
+    pub struct DataLocation: u16 {
+        ///
+        const BUFFER = 0b00000001;
+        ///
+        const SNAPSHOT = 0b00000010;
+        ///
+        const PERSISTING = 0b00000100;
+        ///
+        const BUFFER_SNAPSHOT = 0b0001000;
+        ///
+        const BUFFER_PERSISTING = 0b0010000;
+        ///
+        const SNAPSHOT_PERSISTING = 0b0100000;
+        ///
+        const BUFFER_SNAPSHOT_PERSISTING = 0b1000000;
+    }
```

Comment:
The point of a bitflags type is that it models a set of booleans. This should be

```suggestion
    pub struct DataLocation: u8 {
        const BUFFER = 0b001;
        const SNAPSHOT = 0b010;
        const PERSISTING = 0b100;
        
        // If you wanted the precomposed aliases:
        const BUFFER_SNAPSHOT = Self::BUFFER.bits | Self:: SNAPSHOT.bits;
        const BUFFER_PERSISTING = Self::BUFFER.bits | Self:: PERSISTING.bits;
        const SNAPSHOT_PERSISTING = Self::SNAPSHOT.bits | Self:: PERSISTING.bits;
        const BUFFER_SNAPSHOT_PERSISTING =  Self::BUFFER.bits | Self::SNAPSHOT.bits | Self:: PERSISTING.bits;
    }
```

Reply:
Ah, I see. Silly me. I did ask myself why we did not use 0b011. Changing it